_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -388,53 +415,17 @@ mfxStatus MFXVideoENCODE_QueryIOSurf(mfxSession session, mfxVideoParam *par, mfx
         else
 #endif
         {
-            switch (par->mfx.CodecId)
-            {
-#ifdef MFX_ENABLE_H264_VIDEO_ENCODE
-            case MFX_CODEC_AVC:
-#if defined (MFX_ENABLE_H264_VIDEO_ENCODE_HW)
-                mfxRes = MFXHWVideoENCODEH264::QueryIOSurf(session->m_pCORE.get(), par, request);
-#else //MFX_VA
-                mfxRes = MFXVideoENCODEH264::QueryIOSurf(par, request);
-#endif //MFX_VA
-                break;
-#endif // MFX_ENABLE_H264_VIDEO_ENCODE
-
-
-#ifdef MFX_ENABLE_MPEG2_VIDEO_ENCODE
-            case MFX_CODEC_MPEG2:
-                mfxRes = MFXVideoENCODEMPEG2_HW::QueryIOSurf(session->m_pCORE.get(), par, request);
-#endif // MFX_ENABLE_MPEG2_VIDEO_ENC
-
-#if defined(MFX_ENABLE_MJPEG_VIDEO_ENCODE)
-            case MFX_CODEC_JPEG:
-                mfxRes = MFXVideoENCODEMJPEG_HW::QueryIOSurf(session->m_pCORE.get(), par, request);
-#endif // MFX_ENABLE_MJPEG_VIDEO_ENCODE
-
-#if defined(MFX_ENABLE_H265_VIDEO_ENCODE)
-            case MFX_CODEC_HEVC:
-            {
 #if defined (MFX_ENABLE_HEVC_VIDEO_FEI_ENCODE)
-                bool * feiEnabled = (bool*)session->m_pCORE->QueryCoreInterface(MFXIFEIEnabled_GUID);//required to check FEI plugin registration.
-                MFX_CHECK_NULL_PTR1(feiEnabled);
-                if(*feiEnabled)
-                    mfxRes = MfxHwH265FeiEncode::H265FeiEncode_HW::QueryIOSurf(session->m_pCORE.get(), par, request);
-                else
+            // required to check FEI plugin registration
```

Comment:
Remove `#ifdef`. Make sure MFXIFEIEnabled_GUID is not covered by any macro to be able to request it always.